### PR TITLE
Added Cache-Control header test and fixed header assertion

### LINKED ISSUE #652: Add tests for `cache-control` header using http_spec
Write tests under `/http` to assert if the cache control headers are being forwarded correctly when set in any upstream responses. Also test if minimum is computed correctly in case multiple requests sent upstream have different cache control headers.

### PR DESCRIPTION
**Pull request by A-N-uraag**
**Summary:**  
Added Cache-Control header test to verify max-age calculation and fixed header assertion

**Issue Reference(s):**  
Fixes #652 
/claim #652 

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [x] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [x] I have added relevant unit & integration tests.
- [ ] I have updated the documentation accordingly (if applicable).
- [x] I have performed a self-review of my own code.

### CODE REVIEW

**Comment by A-N-uraag** [comment]
Hi @tusharmath, the` max-age` calculation is happening correctly when both the headers have max-age. But, I think the test should actually have the expectation to return `no-cache` instead of `max-age`?

File: tests/http_spec.rs [hunk 1]
```diff
@@ -255,7 +255,10 @@ async fn assert_downstream(spec: HttpSpec) {
 
       // Assert Headers
       for (key, value) in assertion.response.0.headers.iter() {
-        assert_eq!(actual_headers.get(key), Some(value));
+        match actual_headers.get(key) {
+          None => panic!("Expected header {} to be present. But not found", key),
```
Comment (tusharmath) [inline tests/http_spec.rs:259]:
```suggestion
          None => panic!("Expected header {} to be present", key),
```
nitpik: just to making it concise.

File: tests/http/cache-control.yml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+          u2:
+            id: 2
+  - request:
+      method: POST
+      url: http://localhost:8080/graphql
+      body:
+        query: "query { u1: user(id: 1) { id } u3: user(id: 3) { id } }"
+    response:
+      headers:
+        Cache-Control: "max-age=3600" # BUG: This should be 'no-store'
```
Comment (tusharmath) [inline tests/http/cache-control.yml:83]: Added https://github.com/tailcallhq/tailcall/issues/670

**Review by tusharmath — APPROVED** [review]
